client/src/viya/profile.ts

```diff
@@ -0,0 +1,404 @@
+      pv.error = "No Active Profile";
+      return pv;
+    }
+    pv.profile = profileDetail.profile;
+    if (profileDetail.profile['client-id']) {
+      pv.type = AuthType.Password;
+    }
+    else if (profileDetail.profile['token-file']) {
+      pv.type = AuthType.TokenFile;
+      try {
+        pv.data = readFileSync(profileDetail.profile['token-file'], 'utf-8');
+      } catch (err) {
+        pv.error = `Please update profile (${profileDetail.name}): ${err.message}`;
+        pv.type = AuthType.Error;
+      }
+    } else {
+      pv.error = "No token or client found";
+    }
+    return pv;
+  }
+
+  /**
+   * Upsert allows for add or update the new {@link Profile} into the current profile confiugration.
+   * 
+   * @param name {@link String} of the name of the profile
+   * @param profile {@link Profile} object
+   */
+  async upsertProfile(name: string, profile: Profile): Promise<void> {
+    this.sanitize(profile);
+    this.value[name] = profile;
+    await this.update(this.value);
+  }
+
+  /**
+   * Deletes a profile from the profile configuration.
+   * 
+   * @param name {@link String} of the name of the profile
+   */
+  async deleteProfile(name: string): Promise<void> {
+    if (name in this.value) {
+      if (this.value[name]['active']) {
+        this.updateActiveProfile();
+      }
+      delete this.value[name];
+      await this.update(this.value);
+    }
+  }
+
+  /**
+   * Updates the current active profile in the profile configuration by prompting
+   * the user for updated information.  The user prompts will provide the current
+   * object values to allow the user override any values needed to change.
+   */
+  async updateActiveProfile(): Promise<void> {
+    const profileList = this.value;
+    if (await this.length() === 0) {
+      const newProfile = await createInputTextBox(ProfilePromptType.NewProfile);
+      await this.prompt(newProfile);
+    }
+    await this.update(profileList);
+  }
+
+  /**
+   * Requests users input on updating or adding a new profile.
+   * 
+   * @param name the {@link String} represntation of the name of the profile
+   * @param forceUpdate the {@link Boolean} of whether to prompt the user when value is already defined
+   */
+  async prompt(name: string, forceUpdate = false): Promise<void> {
+    const profile = name in this.value ? this.value[name] : <Profile>{ "sas-endpoint": "", "client-id": "", "client-secret": "", "compute-context": "", "active": false }
+
+    if (!profile['sas-endpoint'] || forceUpdate) {
+      profile['sas-endpoint'] = await createInputTextBox(ProfilePromptType.HostName, profile['sas-endpoint']);
+    }
+    if (!profile['compute-context'] || forceUpdate) {
+      profile['compute-context'] = DEFAULT_COMPUTE_CONTEXT;
+      profile['compute-context'] = await createInputTextBox(ProfilePromptType.ComputeContext, profile['compute-context']);
+    }
+    if ((!profile['client-id'] || !profile['token-file']) || forceUpdate) {
```

Comment:
Just an idea would it be better to let user choose from password flow and token flow and prompt different things? Or, perhaps we can prompt inline when it can leave empty

Reply:
We discussed this prompt, we decided to not include it as it seems like a more simplistic workflow to have the user not choose from a dropdown list, but rather updated the README to let the user know their flow before attempting. Please let me know your thoughts.

Reply:
That's a good recommendation!

Reply:
implemented